src/jobs_server.erl

```diff
@@ -89,7 +89,7 @@ ask() ->
     ask(default).
 
 
--spec ask(job_class()) -> {ok, reg_obj()} | {error, rejected | timeout}.
+-spec ask(job_class()) -> {ok, counter()} | {error, rejected | timeout}.
```

Comment:
Actually, the return type is not `{ok, counter()}`, but I can understand if dialyzer accepts it.
It should, IMHO, be `{ok, reg_obj()}`, where `reg_obj() :: {reference(), [{info, any()} | {counters, [counter()]}]}`, and where `counters() :: {#cr{}, non_neg_integer()}`

Several compounding type errors in the old code, that is.